### ISSUE TITLE
 Could not find class [org.springframework.context.annotation.ScopedProxyMode] for OAuth2AutoConfiguration

### ISSUE THREAD

**Issue by VuDo98** [spring-projects-experimental/spring-native#618]
I have 2 dependencies `spring-boot-starter-security` and `spring-security-oauth2-autoconfigure` which is used for OAuth2. The project built ok but got error at startup. Not sure if I need a demo for this.

```
org.springframework.beans.factory.BeanDefinitionStoreException: Failed to process import candidates for configuration class [org.springframework.boot.autoconfigure.security.oauth2.OAuth2AutoConfiguration]; nested exception is java.lang.IllegalArgumentException: Could not find class [org.springframework.context.
annotation.ScopedProxyMode]
        at org.springframework.context.annotation.ConfigurationClassParser.processImports(ConfigurationClassParser.java:610)
        at org.springframework.context.annotation.ConfigurationClassParser.doProcessConfigurationClass(ConfigurationClassParser.java:311)
        at org.springframework.context.annotation.ConfigurationClassParser.processConfigurationClass(ConfigurationClassParser.java:250)
        at org.springframework.context.annotation.ConfigurationClassParser.processImports(ConfigurationClassParser.java:600)
        at org.springframework.context.annotation.ConfigurationClassParser.access$800(ConfigurationClassParser.java:111)
        at org.springframework.context.annotation.ConfigurationClassParser$DeferredImportSelectorGroupingHandler.lambda$processGroupImports$1(ConfigurationClassParser.java:812)
        at java.util.ArrayList.forEach(ArrayList.java:1259)
        at org.springframework.context.annotation.ConfigurationClassParser$DeferredImportSelectorGroupingHandler.processGroupImports(ConfigurationClassParser.java:809)
        at org.springframework.context.annotation.ConfigurationClassParser$DeferredImportSelectorHandler.process(ConfigurationClassParser.java:780)
        at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:193)
        at org.springframework.context.annotation.ConfigurationClassPostProcessor.processConfigBeanDefinitions(ConfigurationClassPostProcessor.java:330)
        at org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanDefinitionRegistry(ConfigurationClassPostProcessor.java:246)
        at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanDefinitionRegistryPostProcessors(PostProcessorRegistrationDelegate.java:311)
        at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:112)
        at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:745)
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:563)
        at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:144)
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:767)
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:759)
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:426)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:326)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1311)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1300)
        at com.inspira.socket.SocketApplicationServer.main(SocketApplicationServer.java:28)
Caused by: java.lang.IllegalArgumentException: Could not find class [org.springframework.context.annotation.ScopedProxyMode]
        at org.springframework.util.ClassUtils.resolveClassName(ClassUtils.java:334)
        at org.springframework.core.type.classreading.MergedAnnotationReadingVisitor.visitEnum(MergedAnnotationReadingVisitor.java:104)
        at org.springframework.core.type.classreading.MergedAnnotationReadingVisitor.visitEnum(MergedAnnotationReadingVisitor.java:80)
        at org.springframework.asm.ClassReader.readElementValue(ClassReader.java:3056)
        at org.springframework.asm.ClassReader.readElementValues(ClassReader.java:2975)
        at org.springframework.asm.ClassReader.readMethod(ClassReader.java:1390)
        at org.springframework.asm.ClassReader.accept(ClassReader.java:721)
        at org.springframework.asm.ClassReader.accept(ClassReader.java:401)
        at org.springframework.core.type.classreading.SimpleMetadataReader.<init>(SimpleMetadataReader.java:49)
        at org.springframework.core.type.classreading.SimpleMetadataReaderFactory.getMetadataReader(SimpleMetadataReaderFactory.java:103)
        at org.springframework.boot.type.classreading.ConcurrentReferenceCachingMetadataReaderFactory.createMetadataReader(ConcurrentReferenceCachingMetadataReaderFactory.java:86)
        at org.springframework.boot.type.classreading.ConcurrentReferenceCachingMetadataReaderFactory.getMetadataReader(ConcurrentReferenceCachingMetadataReaderFactory.java:73)
        at org.springframework.core.type.classreading.SimpleMetadataReaderFactory.getMetadataReader(SimpleMetadataReaderFactory.java:81)
        at org.springframework.context.annotation.ConfigurationClassParser.asSourceClass(ConfigurationClassParser.java:696)
        at org.springframework.context.annotation.ConfigurationClassParser$SourceClass.getMemberClasses(ConfigurationClassParser.java:993)
        at org.springframework.context.annotation.ConfigurationClassParser.processMemberClasses(ConfigurationClassParser.java:355)
        at org.springframework.context.annotation.ConfigurationClassParser.doProcessConfigurationClass(ConfigurationClassParser.java:272)
        at org.springframework.context.annotation.ConfigurationClassParser.processConfigurationClass(ConfigurationClassParser.java:250)
        at org.springframework.context.annotation.ConfigurationClassParser.processImports(ConfigurationClassParser.java:600)
        ... 23 common frames omitted
Caused by: java.lang.ClassNotFoundException: org.springframework.context.annotation.ScopedProxyMode
        at com.oracle.svm.core.hub.ClassForNameSupport.forName(ClassForNameSupport.java:60)
        at java.lang.Class.forName(DynamicHub.java:1260)
        at org.springframework.util.ClassUtils.forName(ClassUtils.java:284)
        at org.springframework.util.ClassUtils.resolveClassName(ClassUtils.java:324)
        ... 41 common frames omitted

```

**Comment by VuDo98** [comment]
here is a demo.
[src.zip](https://github.com/spring-projects-experimental/spring-native/files/6144968/src.zip)

**Comment by VuDo98** [comment]
I believe creating a hint for this is bad because it is a proxy. Need you guys to create it at build time

**Comment by eleftherias** [comment]
Thanks for reaching out @VuDo98.

It seems like you are using the legacy Spring Security OAuth project.

Are you aware that that project in now [in maintenance mode](https://spring.io/blog/2020/05/07/end-of-life-for-spring-security-oauth)?
That means that we are not adding any additional features to it.

I would encourage you to switch to Spring Security 5 which has integrated support for the OAuth 2.0 client and resource server.

If you have trouble switching or if there is a reason why you cannot switch, please reach out to me here.

**Comment by VuDo98** [comment]
hi @eleftherias. In my demo, I didn't add anything config other than having spring auto config oauth2 thru `spring-security-oauth2-autoconfigure`. If I want to switch does that means I have to remove this dep? Also, We have a legacy custom authorization server. But I don't think its having anything to do with oauth2 client.

**Comment by eleftherias** [comment]
@VuDo98 `spring-security-oauth2-autoconfigure` is in [maintenance mode](https://docs.spring.io/spring-security-oauth2-boot/docs/current/reference/html5/).

If you are using that dependency to make your application an OAuth 2 client or OAuth 2 resource server, you should check out Spring Security’s [feature matrix](https://github.com/spring-projects/spring-security/wiki/OAuth-2.0-Features-Matrix) to see if the new first-class support meets your needs.

There is [not support](https://spring.io/blog/2019/11/14/spring-security-oauth-2-0-roadmap-update) for the OAuth 2 authorization server in Spring Security 5, but we are working on an experimental [Spring Authorization Server](https://github.com/spring-projects-experimental/spring-authorization-server) project.

In summary, we are not adding native support to the `spring-security-oauth2-autoconfigure` project because it is in maintenance mode.
